src/api/csrfAuthenticator.js

```diff
@@ -0,0 +1,39 @@
+var logger = require('../logger')(3);
+
+var CSRF_HEADER = 'x-csrf-token';
+
+function CsrfAuthenticator(options){
```

Comment:
I am also fine with declarative setup like request.authenticate({crsf: {user,pass,url})

Reply:
there is no user or pass that could be associated with csrf

Reply:
There is for sure, the call for getting the crsf token should be authenticated, see the reference impl: https://github.wdf.sap.corp/I533952/uiveri5_iscoper/blob/9f08dcbbf05b692184aa619266ea1b3904ad8e61/iscoperPlugin.js#L43

Reply:
thanks for this link but as you know I've had it for 2 months and looked at it every day since.
please read a bit more about csrf, bearer tokens and relaystate param (as it also came up) and that they are not the same e.g.
https://owasp.org/www-community/attacks/csrf
https://portswigger.net/web-security/csrf/tokens
https://portswigger.net/web-security/csrf
https://blogs.sap.com/2019/02/19/what-is-relaystate-in-saml-and-how-to-configure-relaystate-on-as-abap/
https://stackoverflow.com/questions/25838183/what-is-the-oauth-2-0-bearer-token-exactly/25843058

if you want to add support for authentication, it would be a separate topic. besides, user:auth in the url is accepted https://visionmedia.github.io/superagent/#authentication